From 57bec8eecf41d66bb9644874baa9190a73f9843a Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 10 Apr 2024 10:50:44 +0200
Subject: [PATCH 561/841] fix(stm32mp2): increase BL2 DTB size

Due to some recent updates in DT (DDR supplies, cryp peripherals),
the DTB is now bigger that 20kB when compiled with TRUSTED_BOARD_BOOT.
We can safely and unconditionally increase it by 4kB, as it is placed
before BL2 and is unmapped before loading BL31.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 6564fcedf1317e04c531ab4b75244e6aa7b6cc9e fix(stm32mp2): increase BL2 DTB size]
Change-Id: I1c6b40823d8ee3031ad4ebbbe63ca65ce40d4acd
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/370279
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/372642
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index b9e2887e3..ff3fd7255 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -267,7 +267,7 @@ enum ddr_type {
 #endif
 
 /* DTB initialization value */
-#define STM32MP_BL2_DTB_SIZE			U(0x00005000) /* 20 KB for DTB */
+#define STM32MP_BL2_DTB_SIZE			U(0x00006000)	/* 24 KB for DTB */
 
 #define STM32MP_BL2_DTB_BASE			(STM32MP_BL2_BASE - \
 						 STM32MP_BL2_DTB_SIZE)
-- 
2.47.2

